### PR TITLE
number_distributed_tags: sort tags by repr

### PR DESCRIPTION
**Pull request by matthiasdiener**
cc @MTcam

Should resolve issues like
```
TypeError: '<' not supported between instances of 'type' and 'type'
```

Followup to #462.

### CODE REVIEW

File: pytato/distributed/tags.py [hunk 1]
```diff
@@ -106,7 +106,7 @@ def set_union(
         next_tag = base_tag
         assert isinstance(all_tags, frozenset)
 
-        for sym_tag in sorted(all_tags):
+        for sym_tag in sorted(all_tags, key=lambda tag: repr(tag)):
```
Comment (inducer) [inline pytato/distributed/tags.py:109]: This now requires that tags have a textual representation... so please update the docs.

**Comment by kaushikcfd** [comment]
The repr should be consistent too. I don't think `Record.__repr__` is same
across interpreter runs.

**Comment by matthiasdiener** [comment]
> The repr should be consistent too. I don't think `Record.__repr__` is same across interpreter runs.

Yes, that's true, and also applies to frozenset etc. Between this approach and #469, I'm not sure which one is better.

File: pytato/distributed/tags.py (hunk 1 above)
Comment (matthiasdiener) [reply pytato/distributed/tags.py:110]: Done in [2c9ad50](https://github.com/inducer/pytato/pull/467/commits/2c9ad5086b009fb2c0facf849c273a338c67d021).

**Comment by matthiasdiener** [comment]
Closing in favor of #469.

**Pull request closed without merging**